src/main/java/hudson/remoting/Launcher.java

```diff
@@ -404,26 +431,18 @@ public void run() throws CmdLineException, IOException, InterruptedException {
         }
         workDirManager.setupLogging(internalDirPath, agentLog != null ? PathUtils.fileToPath(agentLog) : null);
 
+        // Initialize certificates
         createX509Certificates();
         try {
             sslSocketFactory = Engine.getSSLSocketFactory(x509Certificates, noCertificateCheck);
         } catch (GeneralSecurityException | PrivilegedActionException e) {
             throw new RuntimeException(e);
         }
-        if(connectionTarget!=null) {
-            runAsTcpClient();
-        } else
-        if (agentJnlpURL != null || !urls.isEmpty() || directConnection != null) {
-            if (agentJnlpURL != null) {
-                bootstrapInboundAgent();
-            }
-            runAsInboundAgent();
-        } else
-        if(tcpPortFile!=null) {
-            runAsTcpServer();
-        } else {
-            runWithStdinStdout();
+        if (noCertificateCheck) {
+            hostnameVerifier = new NoCheckHostnameVerifier();
```

Comment:
Initializing this eagerly rather than lazily for consistency with other code. Of course, this necessitates that all public methods in this file now call initialization, which is why I factored out initialization into its own method.